web/src/main/kotlin/com/example/ajaxproject/service/PrivateChatServiceImpl.kt

```diff
@@ -13,36 +13,36 @@ import org.bson.types.ObjectId
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @Service
 class PrivateChatServiceImpl(
     private val privateChatRoomRepository: PrivateChatRoomRepository,
     private val privateChatMessageRepository: PrivateChatMessageRepository,
-    private val userService: UserService,
+    private val userService: UserService
 ) : PrivateChatService {
 
     companion object {
-        private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
+        private val logger: Logger = LoggerFactory.getLogger(PrivateChatServiceImpl::class.java)
     }
 
-    override fun createPrivateRoom(senderId: String, recipientId: String): PrivateChatRoom {
-
-        userService.getUserById(senderId)
-
-        userService.getUserById(recipientId)
-
-        val roomId = roomIdFormat(senderId, recipientId)
-
-        logger.info("Create new private room {}", roomId)
-
-        return privateChatRoomRepository.findChatRoomById(roomId)
-            ?: privateChatRoomRepository.save(
-                PrivateChatRoom(
-                    id = roomId,
-                    senderId = senderId,
-                    recipientId = recipientId
-                )
-            )
+    override fun createPrivateRoom(senderId: String, recipientId: String): Mono<PrivateChatRoom> {
+        return userService.getById(senderId)
+            .then(userService.getById(recipientId))
+            .then(Mono.defer {
+                val roomId = roomIdFormat(senderId, recipientId)
+                privateChatRoomRepository.findChatRoomById(roomId)
+                    .switchIfEmpty(
+                        privateChatRoomRepository.save(
+                            PrivateChatRoom(
+                                id = roomId,
+                                senderId = senderId,
+                                recipientId = recipientId
+                            )
+                        )
+                    )
+            })
     }
```

Comment:
Won't it be the same thing? 

```suggestion
    override fun createPrivateRoom(senderId: String, recipientId: String): Mono<PrivateChatRoom> {
        val roomId = roomIdFormat(senderId, recipientId)

        return privateChatRoomRepository.findChatRoomById(roomId)
            .switchIfEmpty(
                privateChatRoomRepository.save(
                    PrivateChatRoom(
                        id = roomId,
                        senderId = senderId,
                        recipientId = recipientId
                    )
                )
            )
    }
```

Reply:
Made it in another way